<launch>
    <node pkg="the_learner" name="connect_controller" type="connect_controller.sh" output="screen" />
    <node pkg="joy" name="joy_node" type="joy_node"/>
    <node pkg="the_learner" name="sensor_publish_and_control" type="sensor_publish_and_control.py" args="/dev/rfcomm0" output="screen" />
    
    <node pkg="nodelet" type="nodelet" name="imu_manager" 
        args="manager" output="screen" />

    <node pkg="imu_complementary_filter" type="complementary_filter_node"
      name="complementary_filter_gain_node" output="screen">
        <param name="do_bias_estimation" value="true"/>
        <param name="do_adaptive_gain" value="true"/>
        <param name="use_mag" value="false"/>
        <param name="gain_acc" value="0.05"/>
    </node>

    <node pkg="the_learner" name="tf_broadcaster" type="tf_broadcaster.py"/>
    <node pkg="tf" type="static_transform_publisher" name="odom_to_map"
      args="0 0  0 0 0 0  /map /learner/odom 10" />
    <node pkg="the_learner" name="state_estimator" type="state_estimator.py" output="screen" />

    <node pkg="the_learner" name="map_server" type="map_server.py" output="screen">
        <param name="mapping_sensors" value="/sensors/ultrasonic_0,/sensors/ultrasonic_1,/sensors/ultrasonic_2,/sensors/ultrasonic_3,/sensors/ultrasonic_4,/sensors/ultrasonic_5,/sensors/ultrasonic_6"/>
    </node>
</launch>
